### ISSUE TITLE
Meaning of the Sum of Shapley Values

### ISSUE THREAD

**Issue by FabianSchuetze** [shap/shap#392]
Can somebody kindly explain to me what the sum of the Shapley values represent? 

 I used the following code to predict whether a handwritten digit equals `5` with lightgbm. I then create an instance of the `shap.tree_explainer` object with the following arguments: `model_output: probability` and `feature_dependence = independent`. From reading the[ documentation](https://shap.readthedocs.io/en/latest/) I understand the following: 

> If “probability” then we explain the output of the model transformed into probability space (note that this means the SHAP values now sum to the probability output of the model)` 

and 

> Currently the probability and log_loss options are only supported when feature_dependence=”independent”.`.

However, when I calculate shap_values, the maximum shape value is 19 whilst the maximum prediction of the lightgbm model is 1. Can somebody kindly explain to me what the shap_values mean in this instance here? I seem to misunderstand the documentation. 

Here is the code I used:

    from sklearn.datasets import load_digits
    import lightgbm as lbm
    import shap

    digits = load_digits()
    X = digits['data']
    Y = digits['target']
    Y = (Y == 5).astype(int)
    dtrain = lbm.Dataset(X, Y)

    PARAM = {'task':'train', 'boosting_type': 'gbrt',
             'objective': 'binary',
             'metric': 'auc',
             'verbose':-1}

    bst = lbm.train(PARAM, dtrain, num_boost_round=100, verbose_eval=False)
    explainer = shap.TreeExplainer(bst, model_output='probability',
                                   feature_dependence='independent')
    shap_values = explainer.shap_values(X)

    print('The min and max predicted values are %f %f'\
            %(bst.predict(X).min(), bst.predict(X).max()))

    print('The max of the sum of shapley_values is %f'\
            %shap_values.sum(axis=1).max())

The issue might be related to issue #168

**Comment by slundberg** [comment]
That `model_output` option is a new feature, and it depends on knowing the objective of the model. When loading LightGBM models getting the objective had trouble which I just fixed (sorry it was causing problems!). So if you run the latest master the following code now works:
```
from sklearn.datasets import load_digits
import lightgbm as lbm
import shap
import sklearn
import numpy as np

digits = load_digits()
X = digits['data']
Y = digits['target']
Y = (Y == 5).astype(int)
dtrain = lbm.Dataset(X, Y)

PARAM = {'task':'train', 'boosting_type': 'gbrt',
         'objective': 'binary',
         'metric': 'auc',
         'verbose':-1}

bst = lbm.train(PARAM, dtrain, num_boost_round=100, verbose_eval=False)
background = sklearn.utils.resample(X, n_samples=100, replace=False)
explainer = shap.TreeExplainer(bst, background, model_output='probability',
                               feature_dependence='independent')
shap_values = explainer.shap_values(X)

# this outputs essentially zero
print(np.abs(explainer.expected_value + shap_values.sum(axis=1) - bst.predict(X)).max())
```

**Comment by FabianSchuetze** [comment]
Thank you very much.
